### PR TITLE
diswall: init at 0.2.0

### PR DESCRIPTION
**Pull request by Izorkin**
###### Description of changes
Add diswall package version 0.1.12

cc @SuperSandro2000

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by misuzu** [comment]
AFAIK `diswall` uses `ipset` which is not compatible with `nftables` backend used in NixOS. There is an [issue](https://github.com/dis-works/diswall-rs/issues/2) about `nftables` support though.

**Comment by Izorkin** [comment]
Update to 0.2.0.
cc @SuperSandro2000

File: pkgs/applications/networking/diswall/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{ lib, rustPlatform, fetchFromGitHub }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "diswall";
+  version = "0.2.0";
```
Comment (misuzu) [inline pkgs/applications/networking/diswall/default.nix:5]: https://github.com/dis-works/diswall-rs/commit/0509bf696c5442ccf91cddaeaeb70de9560c85df#diff-94da6253c6e14901145d97aaaa22be83023353976134f4ef9293bf4eb5e257c7R116
Would this even work on NixOS? `/etc/systemd/system` is a symlink to `/nix/store` which is read only.

**Comment by Izorkin** [comment]
Haven't tested yet. I haven't get there to creating a diswall service yet.

File: pkgs/applications/networking/diswall/default.nix (hunk 1 above)
Comment (Izorkin) [reply pkgs/applications/networking/diswall/default.nix:5]: Haven't tested yet. I haven't get there to creating a diswall service yet.

**Pull request merged**